src/class/FStar.Class.PP.fst

```diff
@@ -81,6 +81,10 @@ instance pp_tuple6
             parens (separate comma [pp x1; pp x2; pp x3; pp x4; pp x5; pp x6]));
 }
 
-let from_showable (a:Type) {| _ : Class.Show.showable a |} : Tot (pretty a) = {
+let pretty_from_showable (#a:Type) {| _ : Class.Show.showable a |} : Tot (pretty a) = {
    pp = (fun x -> arbitrary_string (Class.Show.show x));
 }
+
+let showable_from_pretty (#a:Type) {| _ : pretty a |} : Tot (Class.Show.showable a) = {
+   show = (fun x -> render (pp x));
+}
```

Comment:
Is it an issue that there is a cycle in the type class instances now?  Can this cause resolution to loop?

The `pretty_from_showable` instance also feels a bit dangerous, it makes it easy to accidentally get a nonpretty string even if you call `pp`.

Reply:
Oh, you're right!  I totally missed that.